### PR TITLE
Update mapreduce docs

### PR DESCRIPTION
**Pull request by rafilong**
Resolved issues with mapreduce docs and did various cleanup.

https://github.com/illinois-cs241/assignments/issues/894
https://github.com/illinois-cs241/assignments/issues/767

### CODE REVIEW

File: _docs/mapreduce.md [hunk 1]
```diff
@@ -125,23 +125,24 @@ You'll start up one instance of splitter for each mapper, using a pipe to send `
...
 
+Sample Usage:
 
 `$ ./mapreduce test.in test.out ./my_mapper ./my_reducer 3`
 
 
 Your program will:
 
-* Split the input file into <mapper_count> parts and pipe the contents into <mapper_count> different mapper processes (use splitter).
+* Split the input file into \<mapper_count\> parts and pipe the contents into \<mapper_count\> different mapper processes (use splitter).
```
Comment (HungWoeiNeoh) [inline _docs/mapreduce.md]: Suggestion: use teletype for `mapper_count`

File: _docs/mapreduce.md [hunk 2]
```diff
@@ -125,23 +125,24 @@ You'll start up one instance of splitter for each mapper, using a pipe to send `
...
+* Pipe the output of the mapper processes into the reduce process
```
Comment (HungWoeiNeoh) [inline _docs/mapreduce.md]: Nit: reducer

File: _docs/mapreduce.md [hunk 3]
```diff
@@ -154,21 +155,25 @@ This too can be done in the Unix shell:
 * Different types of mapper and reducer tasks
 * Both mapper and and reducer generating accurate output to stdout file descriptor independently
 * Splitter being used correctly to generate equally sized input data for each mapper
-* All mappers being run in parallel resulting in at least 2x performance speedup for the pi executable
+* Correctly logging output, including printing non-zero return codes (see `core/utils.h` for helper functions!)
```
Comment (HungWoeiNeoh) [inline _docs/mapreduce.md]: 👍

File: _docs/mapreduce.md (hunk 1 above)
Comment (rafilong) [reply _docs/mapreduce.md]: Teletype?

**Comment by rafilong** [comment]
Fixed HW's nit

File: _docs/mapreduce.md (hunk 1 above)
Comment (HungWoeiNeoh) [reply _docs/mapreduce.md]: Basically, put it as `mapper_count` instead of mapper_count

Comment (rafilong) [reply _docs/mapreduce.md]: Can do

**Pull request merged**